libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1007,9 +1070,10 @@ string ABIFunctions::abiEncodingFunctionStringLiteral(
 
 		if (_to.isDynamicallySized())
 		{
+			solAssert(_to.category() == Type::Category::Array, "");
 			Whiskers templ(R"(
 				function <functionName>(pos) -> end {
-					mstore(pos, <length>)
+					pos := <storeLength>(pos, <length>)
 					<#word>
 						mstore(add(pos, <offset>), <wordValue>)
```

Comment:
Do you think it'd be better to assign ``pos := add(pos, <wordSize>)`` after this ``mstore``, where ``<wordSize>`` is always 32 except for the last element in the non-padded case where it's ``value.size()&31``? Then below it could just be ``end := pos`` and we have less and smaller literal numbers in the code... But not strictly speaking related to this PR and we could keep it as is for now.

Not sure whether the optimizer would add everything up to the same offsets in the end anyways - but I guess that direction is easier to do later than going back to incrementing in small steps - I'm not sure.
All of that optional for this PR in any case.

Reply:
Do we have/Should we have an issue for looking into the optimizer generating large constants by "simplifying" that might be more expensive than repeated increments?

Reply:
The optimizer will add everything up anyway. I think the question is rather what is easier to verify in this source file. I don't think there is such an issue, and I'm also not sure whether we should create issues about all these things...

Reply:
Yeah... it's a lot of issues already... alright!